packages/input-option/src/InputOption/InputOption.style.ts

```diff
@@ -2,77 +2,97 @@ import { css } from '@leafygreen-ui/emotion';
-  'input-option-description',
-);
-
-export const inputOptionStyles = css`
-  position: relative;
-  list-style: none;
-  outline: none;
-  border: unset;
-  margin: 0;
-  text-align: left;
-  text-decoration: none;
-  cursor: pointer;
-
-  font-size: ${typeScales.body1.fontSize}px;
-  line-height: ${typeScales.body1.lineHeight}px;
-  font-family: ${fontFamilies.default};
-  padding: ${spacing[2]}px ${spacing[2] + spacing[1]}px;
-
-  transition: background-color ${transitionDuration.default}ms ease-in-out;
-
-  &:focus,
-  &:focus-visible {
+import { leftGlyphClassName, titleClassName } from '../InputOptionContent';
+
+export const inputOptionClassName = createUniqueClassName('input_option');
+
+interface InputOptionStyleArgs {
+  theme: Theme;
+  disabled?: boolean;
+  highlighted?: boolean;
+  isInteractive?: boolean;
+}
+
+export const getInputOptionStyles = ({
+  theme,
+  disabled,
+  highlighted,
+  isInteractive,
+}: InputOptionStyleArgs) => {
+  const ixnState = highlighted ? State.Focus : State.Default;
+  return css`
+    position: relative;
+    list-style: none;
     outline: none;
     border: unset;
-  }
-`;
+    margin: 0;
+    text-align: left;
+    text-decoration: none;
+    cursor: pointer;
 
-export const titleSelectionStyles = css`
-  .${titleClassName} {
-    font-weight: bold;
-  }
-`;
+    font-size: ${typeScales.body1.fontSize}px;
+    line-height: ${typeScales.body1.lineHeight}px;
+    font-family: ${fontFamilies.default};
+    padding: ${spacing[200]}px ${spacing[300]}px;
 
-export const inputOptionThemeStyles: Record<Theme, string> = {
-  [Theme.Light]: css`
-    color: ${palette.black};
-  `,
-  [Theme.Dark]: css`
-    color: ${palette.gray.light2};
-  `,
-};
-
-export const inputOptionHoverStyles: Record<Theme, string> = {
-  [Theme.Light]: css`
-    &:hover {
-      outline: none;
-      background-color: ${palette.gray.light2};
-    }
-  `,
-  [Theme.Dark]: css`
-    &:hover {
-      outline: none;
-      background-color: ${palette.gray.dark4};
-    }
-  `,
+    transition: ${transitionDuration.default}ms ease-in-out;
+    transition-property: background-color, color;
+
+    color: ${color[theme].text.primary[ixnState]};
+    background-color: ${color[theme].background.primary[ixnState]};
+
+    ${disabled &&
+    css`
+      cursor: not-allowed;
+      color: ${color[theme].text.disabled[ixnState]};
+    `}
+
+    /* Interactive states */
+    ${isInteractive &&
+    !disabled &&
+    css`
+      /* Hover */
+      &:hover {
+        outline: none;
+        color: ${color[theme].text.primary.hover};
+        background-color: ${color[theme].background.primary.hover};
+
+        .${titleClassName} {
+          color: ${color[theme].text.primary.hover};
+        }
+
+        .${leftGlyphClassName} {
+          color: ${color[theme].icon.primary.hover};
+        }
+      }
+
+      /* Focus (styling handled by highlighted prop) */
+      &:focus,
+      &:focus-visible {
```

Comment:
why do we need both `focus` and `focus-visible`?

Reply:
b/c `:focus-visible` styles don't get applied when programmatically setting focus (i.e. with `element.focus()`)

Reply:
doesn't that mean you can get rid of `focus-visible`? `focus` would override `focus-visible`?

Reply:
oh true. yeah, probably